instrumentation/opentelemetry-instrumentation-elasticsearch/tests/test_elasticsearch.py

```diff
@@ -54,9 +58,37 @@
 # pylint: disable=too-many-public-methods
 
 
-@mock.patch(
-    "elasticsearch.connection.http_urllib3.Urllib3HttpConnection.perform_request"
-)
+def normalize_arguments(doc_type, body=None):
+    if major_version < 7:
+        return {"body": body, "doc_type": doc_type} if body else {"doc_type": doc_type}
+    else:
+        return {"document": body} if body else {}
+
+
+def ElasticsearchFactory(*args, **kwargs):
+    client = Elasticsearch(*args, **kwargs)
+    if major_version == 8:
+        client._verified_elasticsearch = True
+    elif major_version == 7:
+        client.transport._verified_elasticsearch = True
+    return client
+
+class PerformRequestMock:
```

Comment:
Yeah it would be nice to use helpers and avoid all the conditionals here. I don't know what the support policy is, but the EOL dates are:

| Version  | EOL date |
| ------------ | ------------- |
| 1.x  | January 2017  |
| 2.x  | February 2018 |
| 5.x  | March 2019 |
| 6.x | February 2022 |

Source: https://web.archive.org/web/20220703152519/https://www.elastic.co/support/eol

Reply:
Is an alternative to have a generic `helpers_es.py` that applies to 2/5/6/7? My understanding is that we would also avoid the conditionals that way?

Reply:
Moved to helpers, after dropping every version < 6 we just have 2 versions older than 8.